### ISSUE TITLE
Threat reported by API

### ISSUE THREAD

**Issue by panplume** [ValorenWoW/sunstrider-core#5259]
**Describe the bug**
API UnitDetailedThreatSituation() may reports wrong values (5th parameter, the raw value). Tested with paladin and others but may require a full investigation.

**To Reproduce**
Paladin (tested in default prot and holy build from beta Camp Manager)
1. No buff/debuff applied
2. Buff Seal of Righteousness
3. attack a regular mob in open world with Judgement
4. /dump UnitDetailedThreatSituation("player", "target")  or any equivalent way to access the wow API
5. check 5th value (divide by 100)
6. this value should equal to the damage (and it is)

Priest (tested in default holy/shadow build and with an empty talent tree)
1. cast Mind Blast on a target
2. /dump UnitDetailedThreatSituation("player", "target")
3. check 5th value (divide by 100)
4. whatever the talents, the value never match (not 1 for 1 in holy/empty build), not reduced by 25% in shadow build

Druid (level 70 switch to empty talent tree)
1. cast Wrath
2. threat value is off

Mage (level 70 no talent)
1. cast Arcane Blast
2. threat wrong again

**Expected behavior**
threat value for damaging ability should be 1 damage = 1 threat if no threat talent/buff are applied.

TLDR some class have correct threat reported, others not.

**Comment by Jerrythemagicdog** [comment]
Thanks for the issue!
Certain classes have hidden threat reduction or threat increases in TBC, you can read about it here: https://wowwiki-archive.fandom.com/wiki/Threat.

**Comment by panplume** [comment]
True, my bad, I forget to check the cloak enchant.
